### PR TITLE
Create auto apireview

### PR DESCRIPTION
**Pull request by praveenkuttappan**
Create API review automatically when API diff is found

### CODE REVIEW

**Comment by praveenkuttappan** [comment]
@weshaggard : Here is the java PR to create Automatic review. Please let me know if any further change is required here.

File: eng/scripts/Language-Settings.ps1 [hunk 1]
```diff
@@ -216,3 +216,29 @@ function Update-java-CIConfig($pkgs, $ciRepo, $locationInDocRepo, $monikerId=$nu
...
+# function is used to filter packages to submit to API view tool
+function Find-java-Artifacts-For-Apireview($artifactDir, $pkgName = "")
+{
+  Write-Host "Checking for source jar in artifact path $($artifactDir)"
+  # Find all source jar files in given artifact directory
+  $files = Get-ChildItem "${artifactDir}" | Where-Object -FilterScript {$_.Name.EndsWith("sources.jar")}
+  if (!$files)
+  {
+    Write-Host "$($artifactDir) does not have any package"
+    exit(1)
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:230]: It is generally bad form to force exit from a helper function. Instead you should return `$null` and handle at the calling site and exit if that is the entry point script.

File: eng/scripts/Create-APIReview.ps1 [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        $StatusCode = $_.Exception.Response.StatusCode
+    }
+
+    return $StatusCode
+}
+
+
+. (Join-Path $PSScriptRoot Language-Settings.ps1)
+$packages = @{}
+if (Test-Path "Function:Find-java-Artifacts-For-Apireview")
```
Comment (weshaggard) [inline eng/scripts/Create-APIReview.ps1:56]: When you move this to common you should also move the name of this function to common.ps1, see https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/scripts/common.ps1#L33.

File: eng/scripts/Create-APIReview.ps1 [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    catch
+    {
+        $StatusCode = $_.Exception.Response.StatusCode
+    }
+
+    return $StatusCode
+}
+
+
+. (Join-Path $PSScriptRoot Language-Settings.ps1)
```
Comment (weshaggard) [inline eng/scripts/Create-APIReview.ps1:54]: This will want to include common.ps1 once you move to eng/common.

**Review by weshaggard — APPROVED** [review]
A couple comments but looks good otherwise.

File: eng/scripts/Language-Settings.ps1 (hunk 1 above)
Comment (praveenkuttappan) [reply eng/scripts/Language-Settings.ps1]: Fixed

**Pull request merged**